primary/src/core.rs

```diff
@@ -62,7 +67,7 @@ pub struct Core<PublicKey: VerifyingKey> {
     /// Output all certificates to the consensus layer.
     tx_consensus: Sender<Certificate<PublicKey>>,
     /// Send valid a quorum of certificates' ids to the `Proposer` (along with their round).
-    tx_proposer: Sender<(Vec<CertificateDigest>, Round)>,
+    tx_proposer: Sender<(Vec<CertificateDigest>, Round, Epoch)>,
 
     /// The last garbage collected round.
     gc_round: Round,
```

Comment:
What about `gc_round` ? Don't we want to reset that in case of epoch change?

Reply:
good point!